### DISCUSSION TITLE
user variables

### DISCUSSION THREAD

**Discussion by emanuelmanea** [qutebrowser/qutebrowser#6274] [Ideas] [answered]
Hello,

I'm trying to find a way to have a user defined/controllable counter variable that I can use in commands like below.
- let's say we start with a default value of of the counter
- and with each execution of a binding a user defined function gets executed and return stored in the varible
- you should be able to tie more functions that operate on the counter(to implement increment/decrement)
- and then when executing another binding, this second binding is executed as many times as value of counter

I have searched, but have not found how to implement such a thing.
Is it possible to have such a customization?

Thank you,
Emanuel

**Comment by The-Compiler** [comment, marked as answer]
With `:spawn` and/or [userscripts](https://github.com/qutebrowser/qutebrowser/blob/master/doc/userscripts.asciidoc), you can spawn external processes, which then can do whatever you want them to - including saving some kind of value in a file.
